api/filesystem/filesystem.cpp

```diff
@@ -630,6 +630,41 @@ json moveFile(const json &input) {
     return output;
 }
 
+json movetoTrash(const json &input){
+    json output;
+    if(!helpers::hasRequiredFields(input, {"path"})) {
+        output["error"] = errors::makeMissingArgErrorPayload();
+        return output;
+    }
+    string source = input["path"].get<string>();
+    string username = "";
+
+    #if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
+    os::CommandResult getUserDetails = os::execCommand("whoami");
+    username = getUserDetails.stdOut;
+    username.pop_back();
+    string destination = "/Users/"+username+"/.Trash";
+    #if defined(__linux__)
+    destination = "/home/"+username+"/.local/share/Trash";
+    #endif
+    string command = "mv \"" + source + "\" \"" + destination + "\"";
+    os::CommandResult commandResult = os::execCommand(command);
```

Comment:
Should check if folder actually exists, and exist if not.

On Linux that folder will only exist if the software [trash-cli](https://github.com/andreafrancia/trash-cli/) is installed, which is not in many distributions because it requires Python3 and its tree of dependencies.

Reply:
I guess I add a check for macOS as well as linux by using following command
`if test -d /destination_path; then echo "exist"; fi`
And throw an error message if trash folder not found. Would this work??

Reply:
Yep, perhaps it can be more invasive and actually create the folder (I don't think many users will be annoyed by it, quite the opposite), if not, give a warning as well.

Reply:
Should I create a folder in NEU_CWD names as TrashNeu

Reply:
Let collaborators decide, but I believe you should add the changes into this one.

Reply:
@shalithasuranga @BeyondMagic sir what should we name the folder if trash folder is not found?

Reply:
trash-cli and software alike use their own system, so invading their environment is not a just cause for us if we just move files to it, it is wrong. We either use their own commands or create a separate folder and move them into it; trash-neutralino or something alike should suffice.